src/components/WeatherWidget/WeatherWidget.jsx

```diff
@@ -0,0 +1,137 @@
+import { StyledWrapper, StyledWrapperWeather } from "./WeatherWidget.styles";
+import { useState, useEffect } from "react";
+import getLocation from "../../services/location";
+import getWeather from "../../api/weather";
+import CircularProgress from "@mui/material/CircularProgress";
+import Box from "@mui/material/Box";
+import { Switch, FormControlLabel } from "@mui/material";
+
+const weatherImages = {
+  rainy: {
+    condition: (data) => data.precipitation_probability > 50,
+    image: "rainy-day",
+    description: "Rainy image",
+  },
+  snowy: {
+    condition: (data) => data.snowfall > 0,
+    image: "snow",
+    description: "Snowy image",
+  },
+  windy: {
+    condition: (data) => data.windspeed > 20,
+    image: "wind",
+    description: "Windy image",
+  },
+  sunny: {
+    condition: (data) => data.temperature > 20,
+    image: "sun",
+    description: "Sunny image",
+  },
+  cloud: {
+    condition: () => true,
+    image: "cloud",
+    description: "Cloudy image",
+  },
+};
+
+export const WeatherWidget = () => {
+  const [weatherData, setWeatherData] = useState(null);
+  const [error, setError] = useState(null);
+  const [temperatureUnit, setTemperatureUnit] = useState("C");
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchWeather = async () => {
+      try {
+        const location = await getLocation();
+        const weather = await getWeather(location);
+        setWeatherData(weather);
+      } catch (err) {
+        setError(err.message);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchWeather();
+  }, []);
+
+  const getWeatherImage = () => {
+    if (!weatherData) {
```

Comment:
Shouldn't in that case w provide some information like "Failed to get weather info" or sth like this? So the method `getWeatherImage` itself in that case would simply return `null`.